src/odemis/gui/comp/overlay/world.py

```diff
@@ -1255,31 +1264,136 @@ def _view_to_phys(self):
+        self.v_end_pos = Vec(self.cnvs.phys_to_view(self.p_end_pos, offset))
+        self._calc_edges()
+
     def start_dragging(self, drag, vpos):
         """
         The user can start dragging (creating/editing/moving) the line when the left mouse button is pressed down
         Args:
             drag: hover mode (HOVER_START, HOVER_LINE, HOVER_END)
             vpos: the view coordinates of the mouse cursor once left click mouse event is fired
         """
-        self.drag_v_start_pos = self.drag_v_end_pos = Vec(vpos)
+        self.drag_v_start_pos = Vec(vpos)
         if self.v_start_pos is None:
             self.v_start_pos = self.drag_v_start_pos
         if self.v_end_pos is None:
-            self.v_end_pos = self.drag_v_end_pos
+            self.v_end_pos = self.drag_v_start_pos
 
-    @abstractmethod
-    def on_motion(self, vpos, ctrl_down):
+        if drag == gui.HOVER_START:
+            self._mode = LINE_MODE_EDIT_START
+        elif drag == gui.HOVER_END:
+            self._mode = LINE_MODE_EDIT_END
+        elif drag == gui.HOVER_LINE:
+            self._mode = LINE_MODE_MOVE
+
+        # Other drag modes are allowed, in which case it's up to the child class to handle it
+
+    def on_motion(self, vpos, ctrl_down=False):
         """
         Given that the left mouse button is already pressed down and the mouse cursor is over the line,
         the user can drag (create/edit/move) any tool until the left button is released.
         Args:
             vpos: the view coordinates of the mouse cursor while dragging
-            ctrl_down (boolean): if True, the ctrl key is pressed while dragging and the line
-            is forced to be at one angle multiple of 45 degrees.
+            ctrl_down (boolean): if True, the ctrl key is pressed while dragging
         """
         self.drag_v_end_pos = Vec(vpos)
+        current_pos = Vec(self.cnvs.clip_to_viewport(self.drag_v_end_pos))
+
+        if self._mode == LINE_MODE_MOVE:
+            self._update_moving(current_pos, ctrl_down)
+        elif self._mode == LINE_MODE_EDIT_START:
+            self._update_start_point(current_pos, ctrl_down)
+        elif self._mode == LINE_MODE_EDIT_END:
+            self._update_end_point(current_pos, ctrl_down)
+        else:
+            # Other modes are allowed, in which case it's up to the child class to handle it
+            return
+
+        self._view_to_phys()
+
+    def _update_moving(self, current_pos, _):
+        """ Update view coordinates while moving the line """
+        diff = current_pos - self.drag_v_start_pos
+        self.v_start_pos = self.v_start_pos + diff
+        self.v_end_pos = self.v_end_pos + diff
+        self.drag_v_start_pos = current_pos
+
+    def _update_start_point(self, current_pos, round_angle):
+        """ Update view coordinates while dragging the starting point.
+        round_angle (bool): if True, the ruler is forced to be at one angle multiple of 45 degrees
```

Comment:
round_angle sounds like a round vs a right angle, unfortunately I also cannot think of a better name for this variable.